src/Chapter2.hs

```diff
@@ -349,7 +366,13 @@ Implement a function that returns only the first half of a given list.
 "b"
 -}
 -- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
+firstHalf :: [a] -> [a]
+firstHalf l = sList 0 (length l) l
+    where
+      sList :: Int -> Int -> [a] -> [a]
+      sList i j l
+          | i < (j `quot` 2) = (l !! i) : sList (i+1) j l
+          | otherwise = []
```

Comment:
This function has the same problem with efficiency as the previous one. You can use `take` function in here as well 🙂